src/launchoptions.rs

```diff
@@ -0,0 +1,211 @@
+use std::{
+    fs,
+    fs::{File, OpenOptions},
+    io::{Read, Write},
+    path::PathBuf,
+};
+
+use anyhow::{Context, Result};
+use regex::{Match, Regex};
+use steamid_ng::SteamID;
+use substring::Substring;
+use tracing::Level;
+
+use crate::gamefinder::locate_steam_launch_configs;
+
+pub const TF2_GAME_ID: &str = "440";
+pub const CONDEBUG: &str = "-condebug";
+pub const CONCLEARLOG: &str = "-conclearlog";
+pub const USERCON: &str = "-usercon";
+pub const G15: &str = "-g15";
+pub fn get_required_args_vec() -> Vec<String> {
+    let args: &[&str; 5] = &[TF2_GAME_ID, CONDEBUG, CONCLEARLOG, USERCON, G15];
+    let mut buf: Vec<String> = Vec::new();
+    args.iter().for_each(|str| buf.insert(0, str.to_string()));
+    buf
+}
+
+pub struct LaunchOptionsV2 {
+    local_config: PathBuf,
+    target_app: String,
+    launch_args_regex: Regex,
+    app_data: Option<String>,
+    new_app_data: Option<String>,
+    required_options: Vec<String>,
+}
+
+/// TODO: Implement a 'write_new_app_data_to_file' fn to write the updated launch options back to the
+/// file specified by 'local_config' struct element.
+/// TODO: Add more tracing to this to be nicer to the user.
```

Comment:
Eh, the tracing is mostly for troubleshooting and debugging, I don't think there's much benefit in trying to log every little detail if it's not really important. 

Ideally the user won't need to even see the logs at all, and if they do need to it would be more helpful to have good error traces (with context) and not an endless list of logs of every little thing the application has been doing.